### ISSUE TITLE
RMSNorm backward broadcasts each row's correction term across columns, so it fails to compile or silently returns wrong input gradients whenever BLOCK_SIZE_BATCH > 1

### ISSUE THREAD

**Issue by truong-v** [BobMcDear/attorch#17]
## Summary

`rms_norm_backward_kernel` multiplies the input by a rank-1 per-row reduction without adding a trailing axis. Triton broadcasts a rank-1 operand along the **last** axis, so the term is applied column-wise: each row picks up a different row's reduction. The LayerNorm sibling writes the same expression correctly, with the `[:, None]` this one omits.

`BLOCK_SIZE_BATCH > 1` whenever `feat_dim < 64` and `batch_dim >= 1024`, and then there are two outcomes. If `BLOCK_SIZE_BATCH != BLOCK_SIZE_FEAT` the shapes clash and the kernel **does not compile**; `tests/utils.py::default_shapes` already contains three such shapes — `(8096, 32)`, `(12288, 1)` and `(2048, 64, 16)` — so `tests/test_rms_norm_layer.py` errors on all of them today. If the two block sizes happen to be equal it compiles and returns **wrong gradients** with no warning; that case is not in `default_shapes`, which is presumably why it went unnoticed.

## Reproduction

```python
import torch
from torch import nn

import attorch

for shape in [(8096, 32), (2048, 2)]:
    torch.manual_seed(0)
    base = torch.randn(shape, device='cuda')
    grad = torch.randn(shape, device='cuda')

    a_in = base.clone().requires_grad_(True)
    t_in = base.clone().requires_grad_(True)
    try:
        attorch.RMSNorm(shape[-1], elementwise_affine=False)(a_in).backward(grad)
    except Exception as error:
        print(shape, type(error).__name__, str(error).splitlines()[0])
        continue
    nn.RMSNorm(shape[-1], elementwise_affine=False, device='cuda')(t_in).backward(grad)
    print(shape, 'input-grad max|diff|', (a_in.grad - t_in.grad).abs().max().item())
```

```
(8096, 32) CompilationError at 83:12:
(2048, 2) input-grad max|diff| 3947.64208984375
```

Controls with `feat_dim >= 64`, where `BLOCK_SIZE_BATCH` is 1 and the broadcast cannot express itself, agree with PyTorch to float noise.

## Environment

- attorch at `88ed34d`
- NVIDIA L40S (sm_89)
- torch 2.4.0+cu121, triton 3.0.0, Python 3.10, driver CUDA 12.4
